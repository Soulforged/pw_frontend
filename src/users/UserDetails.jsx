//@flow
import React from "react";
import { CRUDDetails } from "src/components";

type Props = {
  id: number,
  item: Object,
  openForm: (number) => void,
  fetchUser: (number) => void
};

export default ({
  id, item, openForm, fetchUser
}: Props) => (
  <CRUDDetails
    id={id}
    item={item}
    openForm={openForm}
    loader={fetchUser}
    editButtonTitle="Edit user"
  >
    {item &&
      <div>
        <div className="row">
          <div className="col-sm-3">First name:</div>
          <div className="col-sm-3">{item.firstName}</div>
          <div className="col-sm-3">Last name:</div>
          <div className="col-sm-3">{item.lastName}</div>
        </div>
        <div className="row">
          <div className="col-sm-3">Username:</div>
          <div className="col-sm-3">{item.userName}</div>
          <div className="col-sm-3">Phone number:</div>
          <div className="col-sm-3">{item.mobileNumber}</div>
        </div>
        <div className="row">
          <div className="col-sm-3 col-xs-3">Email:</div>
          <div className="col-sm-3">{item.primaryEmail}</div>
          <div className="col-sm-3">Status:</div>
          <div className="col-sm-3">{item.status}</div>
        </div>
        <div className="row">
          <div className="col-sm-3">Business unit:</div>
          <div className="col-sm-3">{item.businessUnitName}</div>
          <div className="col-sm-3">Role:</div>
          <div className="col-sm-3">{item.roleName}</div>
        </div>
      </div>}
  </CRUDDetails>
);
