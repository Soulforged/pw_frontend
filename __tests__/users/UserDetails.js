//@flow
import React from "react";
import UserDetails from "src/users/UserDetails";
import renderer from "react-test-renderer";

it("details matches snapshot", () => {
  const rendered = renderer.create(<UserDetails fetchUser={jest.fn()} />).toJSON();
  expect(rendered).toMatchSnapshot();
});

it("details matches snapshot with data", () => {
  const rendered = renderer.create(<UserDetails fetchUser={jest.fn()} item={{ id: 1 }} />).toJSON();
  expect(rendered).toMatchSnapshot();
});
