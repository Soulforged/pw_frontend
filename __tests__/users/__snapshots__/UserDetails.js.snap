// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`details matches snapshot 1`] = `
<div
  className="details pnls"
  id="main-pnl"
>
  <div
    className="modal-p details"
  >
    Not found
  </div>
</div>
`;

exports[`details matches snapshot with data 1`] = `
<div
  className="details pnls"
  id="main-pnl"
>
  <div
    className="details"
  >
    <h5
      className="form-header trebuchet text-center bold"
    >
      Details for 
      1
      <button
        className="add-new-btn pointer pull-right bold"
        onClick={[Function]}
        title="Edit"
      >
        <i
          className="edit fa fa-pencil theme"
        />
        Edit user
      </button>
    </h5>
    <div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          First name:
        </div>
        <div
          className="col-sm-3"
        />
        <div
          className="col-sm-3"
        >
          Last name:
        </div>
        <div
          className="col-sm-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          Username:
        </div>
        <div
          className="col-sm-3"
        />
        <div
          className="col-sm-3"
        >
          Phone number:
        </div>
        <div
          className="col-sm-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3 col-xs-3"
        >
          Email:
        </div>
        <div
          className="col-sm-3"
        />
        <div
          className="col-sm-3"
        >
          Status:
        </div>
        <div
          className="col-sm-3"
        />
      </div>
      <div
        className="row"
      >
        <div
          className="col-sm-3"
        >
          Business unit:
        </div>
        <div
          className="col-sm-3"
        />
        <div
          className="col-sm-3"
        >
          Role:
        </div>
        <div
          className="col-sm-3"
        />
      </div>
    </div>
  </div>
</div>
`;
